woocommerce-gateway-dummy.php

```diff
@@ -53,7 +53,11 @@ public static function init() {
 	 * @param array
 	 */
 	public static function add_gateway( $gateways ) {
-		$gateways[] = 'WC_Gateway_Dummy';
+		$options                  = get_option( 'woocommerce_dummy_settings', array() );
+		$hide_for_non_admin_users = $options['hide_for_non_admin_users'];
```

Comment:
This option will not be set for users who are already using the plugin, so it is necessary to add some kind of backwards compatibility check. For example: 

```
		if ( isset( $options[ 'hide_for_non_admin_users' ] ) ) {
			$hide_for_non_admin_users = $options[ 'hide_for_non_admin_users' ];
		} else {
			$hide_for_non_admin_users = 'no';
		}
```

Reply:
Great catch! Added.